data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -483,9 +487,114 @@ public static boolean isForeignKeyWithJoinTable(@NonNull Association association
             }
         }
         createStatements.add(builder.toString());
+        addIndexes(entity, tableName, createStatements);
         return createStatements.toArray(new String[0]);
     }
 
+    private void addIndexes(PersistentEntity entity, String tableName, List<String> createStatements) {
+
+        final String indexes = createIndexesStatementsFromTableAnnotation(entity, tableName);
+        final String indexesFromFields = createIndexesStatementsFromFieldAnnotation(entity, tableName);
+        final String createIndexFromClassAnnotations = createIndexFromClassAnnotations(entity, tableName);
+        if (indexes.length() > 0) {
+            createStatements.add(indexes);
+        } else if (indexesFromFields.length() > 0) {
+            createStatements.add(indexesFromFields);
+        } else if (createIndexFromClassAnnotations.length() > 0) {
+            createStatements.add(createIndexFromClassAnnotations);
+        }
+    }
+
+    private String createIndexFromClassAnnotations(PersistentEntity entity, String tableName) {
+        StringBuilder indexBuilder = new StringBuilder();
+        entity
+                .getAnnotationMetadata()
+                .findDeclaredAnnotation(Indexes.class)
+                .map( x -> x.getAnnotations("value", Index.class))
+                .ifPresent(x -> x.forEach(y ->
+                        addIndex(indexBuilder,
+                                new IndexConfiguration(
+                                        y.getRequiredValue("columnList", String.class),
+                                        y,
+                                        tableName))));
+
+
+        return indexBuilder.toString();
+    }
+
+    private String createIndexesStatementsFromFieldAnnotation(PersistentEntity entity, String tableName) {
+        StringBuilder indexBuilder = new StringBuilder();
+        entity.getPersistentProperties()
+                .stream()
+                .filter(annotatedField -> annotatedField.getAnnotation(Index.class) != null)
+                .forEach(x -> addIndex(indexBuilder, new IndexConfiguration(x
+                        .getAnnotation(Index.class)
+                        .getRequiredValue("columnList", String.class),
+                        x.getAnnotation(Index.class), tableName)));
+        return indexBuilder.toString();
+    }
+
+    private String createIndexesStatementsFromTableAnnotation(PersistentEntity entity, String tableName) {
+
+        StringBuilder indexBuilder = new StringBuilder();
+
+       Optional<AnnotationValue<Annotation>[]> indexes = entity
+                .getAnnotationMetadata()
+                .findDeclaredAnnotation("javax.persistence.Table")
```

Comment:
You should map all the entity `@Table` indexes in the mapper

Reply:
Done

- [ V]